torch/_dynamo/variables/tensor.py

```diff
@@ -677,7 +682,59 @@ def redistribute_fn_with_prim_types(x):
             )
             if not self.source:
                 # Intermediary
-                unimplemented("Intermediary tensors with registered hooks - NYI")
+
+                # This function is only invoked if the passed-in func has no source,
+                # useful for intermediary created functions.
+                def _lifted_fn_src(name):
+                    base = name
+                    for i in itertools.count():
+                        if name not in tx.output.global_scope:
```

Comment:
naw man, you can't do this to me

![image](https://github.com/pytorch/pytorch/assets/13564/0902c207-9667-435f-9a84-b171af9df06f)

Solving this problem in general is the intermediate function reconstruction problem. If you don't need this traversal logic for FSDP, I would much rather you stub this out with an unimplemented

Reply:
Actually I guess I am even more confused, since we're assuming we have the actual python constant at this point, so maybe this isn't even solving the problem I think it's solving?